developer_setup/plugins.md

```diff
@@ -0,0 +1,154 @@
+controllers etc. are loaded during boot time of ManageIQ and are just available as you would expect.
+
+### Cloning plugins
+
+It does not really matter _where_ you clone your plugins. Both approaches have their advantages.
+
+#### Side by side
+
+Having the plugins and the core repository side by side, i.e. at the same directory level, is straight forward and
+lets you separate checkouts cleanly.
+
+```bash
+cd ~/src
+git clone git@github.com:JoeSmith/manageiq-providers-amazon.git
+git clone git@github.com:JoeSmith/manageiq-content.git
+git clone git@github.com:JoeSmith/manageiq-ui-classic.git
+```
+
+#### Inside the core manageiq checkout
+
+The `plugins/` directory inside the core repository is ignored by `.gitignore`. This means you can clone plugins
+into this directory. The benefit is, e.g. when working with a project oriented IDE, you have all code under one
+directory. This way you can easily search across the core _and_ plugin code base. 
+
+You still have to change into the plugin root directory to run the tests - running `rspec` from the manageiq root will
+not pickup your plugin code, but the core code. This is a drawback which can lead to some confusion.
+
+```bash
+cd ~/src/manageiq ; mkdir plugins
+git clone git@github.com:JoeSmith/manageiq-providers-amazon.git plugins/manageiq-providers-amazon
+git clone git@github.com:JoeSmith/manageiq-content.git plugins/manageiq-content
+git clone git@github.com:JoeSmith/manageiq-ui-classic.git plugins/manageiq-ui-classic
+```
+
+### Running tests
+
+To run the tests for a plugin, we need an application context. Usually rails expects a dummy rails app. But because
+our plugins are only for manageiq, we run the tests inside the manageiq app. Therefore we need a checkout of manageiq
+at `spec/manageiq`. The database used for tests is the same as for the core app. Keep that in mind when developing a
+feature that requires database migrations.
+
+```bash
+cd manageiq-providers-amazon
+bin/setup
+# == Cloning manageiq sample app ==
+# Cloning into 'spec/manageiq'...
+bundle install
+bundle exec rake spec:setup
+# ** Preparing database
+# Dropped database 'vmdb_test'
+# Created database 'vmdb_test'
+bundle exec rake
+```
+
+Updating the sample app is handled by `bin/update`:
+
+```bash
+bin/update
+# == Updating manageiq sample app ==
+```
+
+### Cross repository dependencies
+
+Sometimes you need to develop your feature against a branch of manageiq that is not yet merged into master. This creates
+cross repository dependencies, which can be handled from the plugin side or from the core side.
+
+#### Symlink you sample app
+
+Creating a symlink from `spec/manageiq` to your current checkout of manageiq will run the tests inside whatever branch
+you checked out in the core repo.
+
+```bash
+# remove current sample app
+rm -rf spec/manageiq 
+# create a symlink
+ln -s ~/src/manageiq spec/manageiq
```

Comment:
Wherever we have `~/src/manageiq`, let's prefer `/path/to/manageiq`.  Then there is no confusion.  Additionally, down below on line 91 you mentuion override gem, which actually won't work with "~/src" as written